lips/lip-1.mdx

```diff
@@ -107,53 +77,24 @@ Assume two VASPs A and B.  The basic protocol interaction consists of:
-
-Both `_reads` and `_writes` must be exact - meaning that if unused dependencies are specified or insufficient dependencies are specified, the Command will be rejected.
-
-Sample generation of the `cid` can be done with something similar to (in `python3` syntax):
-```
-urandom(16).hex()
-```
+Commands specify the list of objects and versions they expect to operate on in the `_reads` field. The `_reads` field contains a map of object `reference_id` to object versions.  A command is only allowed to read or mutate objects that are referred to within this map. Upon a command being successful all object versions are increased by one. A command failure leaves the object versions unchanged.
 
-<img alt="Object Versioning" src={useBaseUrl('img/object_versioning.png')} />
-
-
-## Protocol Server and Client Roles
-
-In each Channel, one VASP takes the role of a _protocol server_ and the other the role of a _protocol client_ for the purposes of simplifying Shared Object locking / state management. Note that these roles are distinct to the HTTP client/server -- and both VASPs act as an HTTP server and client to listen and respond to requests. To avoid excessive locking and intermediate state management during API requests, by convention the _server_ acts as the source of truth for the state of an Object.  In practice, this means that in the case of lock contention on a Shared Object, the _server_ Command is prioritized.
-
-<img alt="Object Contention" src={useBaseUrl('img/object_contention.png')} />
-
-The protocol server/client role is determined by comparing their binary on-chain Address strings (we call those the _binary address_. The following rules are used to determine which entity serves as which party: The last bit of VASP A’s parent binary address _w_ (where `w = addr[15] & 0x1`) is XOR’d with the last bit in VASP B’s parent binary address _x_.  This results in either 0 or 1. (Refer to https://github.com/libra/libra/tree/master/client/libra-dev#vasp-account-roles for additional information on parent/child VASP accounts).
-If the result is 0, the lexicographically lower parent address is used as the server side.
-If the result is 1, the lexicographically higher parent address is used as the server side. Lexicographic ordering determines which binary address is higher by comparing byte positions one by one, and returning the address with the first higher byte.
+Given the state of an object there is an unambigious way to define who the **next writer** to the object should be (unless the object is in a terminal state and no more commands should be issued on it). Only this VASP is allowed to issue a command to update this Object, and a command from the other VASP will be rejected as a command failure. This prevents concurrent attempts to update Objects that could lead to inconsistencies.
```

Comment:
this is not very clear about what it means of "unambigious way to define who the next writer", 
if I understand correctly, it means the command object (like PaymentObject) design should ensure the unambigious definition of next writer.
may add a ref link to below "PaymentObject Protocol Command Sequence" for example